WordPress/Classes/ViewRelated/NUX/SigninHelpers.swift

```diff
@@ -26,46 +26,94 @@ import Mixpanel
-    class func showLoginFromPresenter(_ presenter: UIViewController, animated: Bool, thenEditor: Bool) {
+    /// Used to present the new login flow from the app delegate
+    fileprivate class func showLoginFromPresenter(_ presenter: UIViewController, animated: Bool, thenEditor: Bool) {
+        defer {
+            trackOpenedLogin()
+        }
+
         let storyboard = UIStoryboard(name: "Login", bundle: nil)
-        if let controller = storyboard.instantiateInitialViewController() {
-            // initial VC is NUXNavigationController
+        if let controller = storyboard.instantiateInitialViewController() as? NUXNavigationController {
             presenter.present(controller, animated: animated, completion: nil)
         }
-        trackOpenedLogin()
     }
 
 
-    // Helper used by the MeViewController
+    /// Used to present the wpcom-only signin flow from the app delegate
     class func showSigninForJustWPComFromPresenter(_ presenter: UIViewController) {
+        if Feature.enabled(.newLogin) {
+            showLoginForJustWPComFromPresenter(presenter)
+            return
+        }
+
         let controller = SigninEmailViewController.controller()
-        controller.restrictSigninToWPCom = true
+        controller.restrictToWPCom = true
 
         let navController = NUXNavigationController(rootViewController: controller)
         presenter.present(navController, animated: true, completion: nil)
 
         trackOpenedLogin()
     }
 
+    /// Used to present the new wpcom-only login flow from the app delegate
+    fileprivate class func showLoginForJustWPComFromPresenter(_ presenter: UIViewController) {
+        defer {
+            trackOpenedLogin()
+        }
+
+        let storyboard = UIStoryboard(name: "Login", bundle: nil)
+        guard let controller = storyboard.instantiateViewController(withIdentifier: "emailEntry") as? NUXAbstractViewController else {
+            return
+        }
+        controller.restrictToWPCom = true
 
-    // Helper used by the BlogListViewController
+        let navController = NUXNavigationController(rootViewController: controller)
+        presenter.present(navController, animated: true, completion: nil)
+    }
+
+
+    /// Used to present the self-hosted signin flow from BlogListViewController
     class func showSigninForSelfHostedSite(_ presenter: UIViewController) {
+        if Feature.enabled(.newLogin) {
+            showLoginForSelfHostedSite(presenter)
+            return
+        }
+
         let controller = SigninSelfHostedViewController.controller(LoginFields())
         let navController = NUXNavigationController(rootViewController: controller)
         presenter.present(navController, animated: true, completion: nil)
 
         trackOpenedLogin()
     }
 
+    /// Used to present the new self-hosted login flow from BlogListViewController
+    fileprivate class func showLoginForSelfHostedSite(_ presenter: UIViewController) {
+        defer {
+            trackOpenedLogin()
```

Comment:
I used the `defer` _because_ of the `guard`, so that the event is tracked regardless of whether that fails or not. Do you think we should only be tracking it if the UI is presented correctly? 🤔